### PR TITLE
Smaato: Rework multi imp support and add adpod support

### PR DESCRIPTION
**Pull request by And1sS**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java [hunk 1]
```diff
@@ -216,14 +160,169 @@ private User modifyUser(User user) {
...
+        return bidRequestBuilder.imp(impSupplier.get()).build();
+    }
+
+    private List<Imp> modifyImpsForAdBreak(List<Imp> imps, String adBreakId) {
+        return IntStream.range(0, imps.size())
+                .mapToObj(idx -> modifyImpForAdBreak(imps.get(idx), idx + 1, adBreakId))
+                .collect(Collectors.toList());
+    }
+
+    private Imp modifyImpForAdBreak(Imp imp, Integer sequence, String tagId) {
```
Comment (nickluck8) [inline src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java:232]: can you rename tagId to adBreakId

File: src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java [hunk 2]
```diff
@@ -233,42 +332,83 @@ private User modifyUser(User user) {
...
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Invalid bid.ext.");
+        }
+    }
+
+    private static int getTtl(MultiMap headers) {
+        try {
+            final long expiresAtMillis = Long.parseLong(headers.get(SMT_EXPIRES_HEADER));
+            final long currentTimeMillis = System.currentTimeMillis();
+            return (int) Math.max((expiresAtMillis - currentTimeMillis) / 1000, 0);
```
Comment (nickluck8) [inline src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java:394]: can you add unit tests for these lines

File: src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java [hunk 3]
```diff
@@ -233,42 +332,83 @@ private User modifyUser(User user) {
...
         final String adMarkupType = headers.get(SMT_ADTYPE_HEADER);
         if (StringUtils.isNotBlank(adMarkupType)) {
             return adMarkupType;
-        }
-        if (adm.startsWith("image")) {
+        } else if (adm.startsWith("{\"image\":")) {
             return SMT_AD_TYPE_IMG;
-        }
-        if (adm.startsWith("richmedia")) {
+        } else if (adm.startsWith("{\"richmedia\":")) {
```
Comment (nickluck8) [inline src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java:406]: unit test for this if

File: src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java [hunk 4]
```diff
@@ -233,42 +332,83 @@ private User modifyUser(User user) {
...
             return SMT_ADTYPE_RICHMEDIA;
-        }
-        if (adm.startsWith("<?xml")) {
+        } else if (adm.startsWith("<?xml")) {
```
Comment (nickluck8) [inline src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java:408]: unit test for this if

File: src/main/java/org/prebid/server/proto/openrtb/ext/request/smaato/ExtImpSmaato.java [hunk 5]
```diff
@@ -16,4 +16,7 @@
 
     @JsonProperty("adspaceId")
     String adspaceId;
+
```
Comment (nickluck8) [inline src/main/java/org/prebid/server/proto/openrtb/ext/request/smaato/ExtImpSmaato.java:19]: remove comment here as well?

**Review by nickluck8 — CHANGES_REQUESTED** [review]

**Review by nickluck8 — APPROVED** [review]

**Review by rpanchyk — APPROVED** [review]

**Pull request merged**